validator/delphi_validator/dynamic.py

```diff
@@ -1,59 +1,52 @@
 """Dynamic file checks."""
 import math
+from dataclasses import dataclass
 from datetime import date, timedelta
+from typing import Dict, Set
 import pandas as pd
 from .errors import ValidationError, APIDataFetchError
 from .datafetcher import get_geo_signal_combos, threaded_api_calls
 from .utils import relative_difference_by_min, TimeWindow
 
 class DynamicValidation:
     """Class for validation of static properties of individual datasets."""
-    def __init__(self, params, suppressed_errors=None):
+
+    @dataclass
+    class Parameters:
```

Comment:
whats the benefit of this over just putting these in the init?

Reply:
We were running into lint issues with the number of class members being too large; it appears that the linter will let us put arbitrarily many variables in a dataclass.

Reply:
ah, interesting. would it be easier to ignore the linter?

Reply:
@krivard thoughts?

Reply:
Doing it in init fails linting because having that many class members is legit complicated and you should look for another way. This usage seems to fit the definition and desired function of a data class. Are there downsides I'm not aware of?

Reply:
My only question is around whether
```
    @dataclass
    class Parameters:
        """Configuration parameters."""
        data_source: str
        time_window: TimeWindow
        generation_date: date
        max_check_lookbehind: timedelta
        smoothed_signals: Set[str]
        expected_lag: Dict[str, int]

    def __init__(self, params):
        """
        Initialize object and set parameters.
        Arguments:
            - params: dictionary of user settings; if empty, defaults will be used
        """
        self.test_mode = params.get("test_mode", False)

        self.params = self.Parameters(
            data_source = params['data_source'],
            time_window = TimeWindow.from_params(params["end_date"], params["span_length"]),
            generation_date = date.today(),
            max_check_lookbehind = timedelta(days=params.get("ref_window_size", 7)),
            smoothed_signals = set(params.get("smoothed_signals", [])),
            expected_lag = params["expected_lag"]
        )

   def foo(self):
       ...
       self.params.test_mode
```
is substantially improved over 
```
    def __init__(self, params):
        """
        Initialize object and set parameters.
        Arguments:
            - params: dictionary of user settings; if empty, defaults will be used
        """
        self.test_mode = params.get("test_mode", False)
        self.data_source = params['data_source']
        self.time_window = TimeWindow.from_params(params["end_date"], params["span_length"])
        self.generation_date = date.today()
        self.max_check_lookbehind = timedelta(days=params.get("ref_window_size", 7))
        self.smoothed_signals = set(params.get("smoothed_signals", []))
        self.expected_lag = params["expected_lag"]

   def foo(self):
       ...
       self.test_mode
```
Maybe once the class grows it becomes more useful?

Reply:
The size growing is the main advantage but a secondary one is that it lets us have other hierarchical information at the class level.  In particular, I'm thinking about a future check enable/disable system that is handled in the `XValidator` classes rather than in the `ValidationReport`.  You would be getting this class initialization:
```
def __init__(self, params):
    self.params = self.Parameters(params)
    self.disabled_checks = self.DisabledChecks(params)
```
rather than
```
def __init__(self, params):
    self.p1 = params["1"]
    ...
    self.pn = params["n"]
    self.enable_check1 = params["check1"]
    ...
    self.enable_checkn = params["checkn"]
```

Reply:
:+1: sgtm